code/modules/projectiles/wavegun.dm

```diff
@@ -30,34 +31,61 @@ toxic - poisons
 */
 	damage_type = D_ENERGY
 	//With what % do we hit mobs laying down
-	hit_ground_chance = 20
+	hit_ground_chance = 33
 	//Can we pass windows
 	window_pass = 0
 
-	on_pointblank(var/obj/projectile/P, var/mob/living/M)
-		stun_bullet_hit(P, M)
 
-/datum/projectile/wavegun/burst
-	shot_number = 3
-	power = 50
-	cost = 75
-	ks_ratio = 0.0
-	damage_type = D_ENERGY
-	sname = "burst wave"
-
-/datum/projectile/wavegun/paralyze
-	icon_state = "wavegun-2"
+/datum/projectile/wavegun/transverse //expensive taser shots that go through /everything/
 	shot_number = 1
-	power = 40
-	ks_ratio = 0.0
+	power = 10 //half the power of a taser at range 1-3, delivers a nasty punch at the 4-tile sweetspot
+	dissipation_delay = 4
+	dissipation_rate = -30
+	max_range = 5 //super short. about 4 tile max range
 	cost = 50
+	hit_ground_chance = 100 //no escape
+	pierces = -1 //no limits
+	goes_through_walls = 1
+	window_pass = 1
+	ticks_between_mob_hits = 1
 	damage_type = D_ENERGY
-	sname = "paralysis wave"
+	sname = "transverse wave"
+	icon_state = "wave-g"
+		
 
-	on_hit(atom/hit)
-		if(isliving(hit))
-			var/mob/living/L = hit
-			L.changeStatus("stunned", 10 SECONDS)
 
-	on_pointblank(var/obj/projectile/P, var/mob/living/M)
-		M.changeStatus("stunned", 10 SECONDS)
+		
+
+/datum/projectile/wavegun/emp
+	shot_number = 1
+	power = 0
+	dissipation_delay = 0
+	dissipation_rate = -10 //only reliable past a few tiles 
+	max_range = 18 //taser-and-a-half range
+	cost = 100 //two shots, unless you upgrade to a pulserifle/etc cell
+	hit_ground_chance = 0
+	damage_type = D_SPECIAL
+	sname = "electromagnetic distruption wave"
+	icon_state = "wave-emp"
+	disruption = 25
+
+	on_hit(atom/H, angle, var/obj/projectile/P)
+		var/turf/T = get_turf(H)
+		if(P.power != 0) //avoid AoE on pointblank... but if you just shoot the guy beside you you're going to get EMPed
+			for(var/turf/tile in range(1,T))
+				for(var/atom/O in tile.contents)
```

Comment:
These two lines could just be `for(var/atom/movable/O in range(1, T))` unless I'm missing something.

Reply:
Guess who didn't know that was a thing. Me.